modular_skyrat/modules/customization/modules/surgery/robot_chassis_restoration.dm

```diff
@@ -53,25 +65,33 @@
 	ironamount = 15
 
 /datum/surgery_step/add_plating/fullbody/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
-	display_results(user, target, span_notice("You begin to add plating to [target]..."),
-			"[user] begins to add plating to [target].",
-			"[user] begins to add plating to [target].")
+	display_results(user, target, span_notice("You begin to add new panels to [target]'s braincase..."),
+			"[user] begins to add new panels to [target]'s braincase.",
+			"[user] begins to add new panels to [target]'s braincase.")
 
-/datum/surgery_step/finalize_chassis_restoration
-	name = "finalize chassis restoration (wrench)"
+/datum/surgery_step/finalize_positronic_restoration
+	name = "finalize positronic restoration (multitool)"
 	implements = list(
-		TOOL_WRENCH = 100)
+		TOOL_MULTITOOL = 100)
+	repeatable = FALSE
 	time = 120
 
-/datum/surgery_step/finalize_chassis_restoration/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
-	display_results(user, target, span_notice("You begin to finalize [target]'s chassis..."),
-			"[user] begins to finalize [target]'s chassis.",
-			"[user] begins to finalize [target]'s chassis.")
+/datum/surgery_step/finalize_positronic_restoration/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	display_results(user, target, span_notice("You begin to force a reboot in [target]'s posibrain..."),
+			"[user] begins to force a reboot in [target]'s posibrain.",
+			"[user] begins to force a reboot in [target]'s posibrain.")
+	target.notify_ghost_cloning("Someone is trying to reboot your posibrain.", source = target)
 
-/datum/surgery_step/finalize_chassis_restoration/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+/datum/surgery_step/finalize_positronic_restoration/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	target.cure_husk()
-	//Revival is handled by REVIVES_BY_HEALING
-	target.heal_overall_damage(1000,1000,0, BODYPART_ROBOTIC)
-	//I call this twice because for some reason health may not update and we may not get a revival, and I cant find a good clean solution to this
-	target.heal_overall_damage(1,1,0, BODYPART_ROBOTIC)
-	return TRUE
+	target.grab_ghost()
+	target.updatehealth()
+	target.setOrganLoss(ORGAN_SLOT_BRAIN, target.getOrganLoss(ORGAN_SLOT_BRAIN) - 200)
+	if(target.revive(full_heal = FALSE, admin_revive = FALSE))
+		target.emote("chime")
+		return TRUE
+	else
+		target.visible_message(span_notice("...[target] reactivates, their chassis coming online!"))
+		to_chat(target, span_danger("[CONFIG_GET(string/blackoutpolicy)]"))
```

Comment:
Correct me if I'm wrong but does this send them the blackout policy when the revival has failed? I got a feeling this is counter-intuitive.

Reply:
It sends them the blackout policy when it works.